plugin.video.101barz/default.py

```diff
@@ -0,0 +1,247 @@
+        for paramsPair in param_pairs:
+            param_splits = paramsPair.split('=')
+            if (len(param_splits)) == 2:
+                param_dict[param_splits[0]] = param_splits[1]
+    return param_dict
+
+
+def add_link(title, url, mode, iconimage, desc="", duration="", youtube_id = ""):
+    u = sys.argv[0] + "?url=" + urllib.parse.quote_plus(url) + "&mode=" + convertToUnicodeString(mode)
+    liz = xbmcgui.ListItem(title, iconImage="DefaultVideo.png", thumbnailImage=iconimage)
+    liz.setInfo(type="Video", infoLabels={"Title": title, "Plot": desc, "Duration": duration})
+    liz.setProperty('IsPlayable', 'true')
+    # let's remove any non-ascii characters from the title, to prevent errors with urllib.parse.parse_qs
+    # of the parameters
+    title = title.encode('ascii', 'ignore')
+    # Add refresh option to context menu
+    liz.addContextMenuItems([('Refresh', 'Container.Refresh')])
+    ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=u, listitem=liz)
+    return ok
+
+
+def add_dir(title, url, mode, iconimage, desc=""):
+    u = sys.argv[0] + '?url=' + urllib.parse.quote_plus(url) + '&mode=' + convertToUnicodeString(mode)
+    liz = xbmcgui.ListItem(title, iconImage='DefaultFolder.png', thumbnailImage=iconimage)
+    liz.setInfo(type='video', infoLabels={'title': title, 'plot': desc, 'plotoutline': desc})
+    # let's remove any non-ascii characters from the title, to prevent errors with urllib.parse.parse_qs
+    # of the parameters
+    title = title.encode('ascii', 'ignore')
+    # Add refresh option to context menu
+    liz.addContextMenuItems([('Refresh', 'Container.Refresh')])
+    ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=u, listitem=liz, isFolder=True)
+    return ok
+
+
+if sys.version_info[0] > 2:
+    unicode = str
+
+
+def convertToUnicodeString(s, encoding='utf-8'):
+    """Safe decode byte strings to Unicode"""
+    if isinstance(s, bytes):  # This works in Python 2.7 and 3+
+        s = s.decode(encoding)
+    return s
+
+
+def convertToByteString(s, encoding='utf-8'):
+    """Safe encode Unicode strings to bytes"""
+    if isinstance(s, unicode):
+        s = s.encode(encoding)
+    return s
+
+
+def log(name_object, object):
+    try:
+        # Let's try and remove any non-ascii stuff first
+        object = object.encode('ascii', 'ignore')
+    except:
+        pass
+
+    try:
+        xbmc.log("[ADDON] %s v%s (%s) debug mode, %s = %s" % (
+            ADDON, VERSION, DATE, name_object, convertToUnicodeString(object)), xbmc.LOGDEBUG)
+    except:
+        xbmc.log("[ADDON] %s v%s (%s) debug mode, %s = %s" % (
+            ADDON, VERSION, DATE, name_object, "Unable to log the object due to an error while converting it to an unicode string"), xbmc.LOGDEBUG)
+
+
+params = parameters_string_to_dict(sys.argv[2])
+mode = params.get('mode')
+url = params.get('url')
+
+if url is None:
+    if SETTINGS.getSetting('onlyshowallvideoscategory') == 'true':
```

Comment:
I think you now have methods to get the setting as bool

Reply:
Will adjust.